### ISSUE TITLE
Fails for responses over 256MB

### ISSUE THREAD

**Issue by batpad** [perliedman/query-overpass#14]
This is likely a bit edge-casey, but JSON.parse on the `body` fails with something like:

```
throw new Error('toString failed');
    ^
Error: toString failed
```

This is likely related to an issue with `node` where strings cannot be larger than 256MB: https://github.com/nodejs/node/issues/3175

A really simple test case shows what's going on: https://github.com/batpad/debug-large-json/blob/master/index.js

The solution would be to read the request as a stream, use a streaming JSON parser to create the json object, and then pass it to `osmtogeojson` here: https://github.com/perliedman/query-overpass/blob/master/index.js#L10

I failed at my streams-fu to get this to work, but filing a ticket here, and I'll try and continue looking for a solution.

(Feel free to close if you feel this does not need to be handled. Just thought it's also an interesting `node` quirk :-) )  - thank you!

**Comment by perliedman** [comment]
Hi and thanks for reporting!

I'd love to see this working, I agree with your assessment that the problem is trying to parse huge JSON strings in one go - streams are clearly the way to go here.

I'm leaving this open, but I doubt I will find the time to address this myself any time soon, since I don't have any real use case at the moment. Help very welcome!

**Comment by tyrasd** [comment]
if you want to give it a try: here's how it is implemented in osmtogeojson's CLI: https://github.com/tyrasd/osmtogeojson/blob/gh-pages/osmtogeojson#L59-L73 (but note that for performance reasons it will fall back to the native JSON.parse() whenever possible)

**Comment by perliedman** [comment]
Although this _should_ be fixed in 1.4.0, please note I haven't tested it, since I was unable to get the Overpass API to give me a response larger than 256MB without timing out.

The module always uses the streaming JSON parser now; since the Overpass API only gives me `Content-Length` in some cases, I couldn't rely on it to know in advance which parser to use.
